extensions/big-data-cluster/src/bigDataCluster/controller/clusterControllerApi.ts

```diff
@@ -150,102 +158,156 @@ export class ClusterController {
 		}
 	}
 
-	public async getEndPoints(): Promise<IEndPointsResponse> {
-		let auth = await this.authPromise;
-		let endPointApi = new BdcApiWrapper(this.username, this.password, this._url, auth);
-		let options: any = {};
+	public async getEndPoints(promptConnect: boolean = false): Promise<IEndPointsResponse> {
 		try {
-			let result = await endPointApi.endpointsGet(options);
-			return {
-				response: result.response as IHttpResponse,
-				endPoints: result.body as EndpointModel[]
-			};
+			return await this.withConnectRetry<IEndPointsResponse>(this.getEndpointsImpl, promptConnect);
```

Comment:
can we also have the try catch blocked into the retry method? you can pass in the error message.